crates/rolldown_sourcemap/src/source.rs

```diff
@@ -83,7 +85,7 @@ impl Source for SourceMapSource {
   }
 
   fn lines_count(&self) -> u32 {
-    self.lines_count
+    self.pre_computed_lines_count.unwrap_or_else(|| lines_count(&self.content))
```

Comment:
@underfin this part is `self.pre_computed_lines_count.unwrap_or(lines_count(&self.content))`. Should be ok now.